common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleBlockResolver.java

```diff
@@ -226,6 +246,29 @@ private void deleteExecutorDirs(String[] dirs) {
     }
   }
 
+  private FilenameFilter filter = new FilenameFilter() {
```

Comment:
Is it feasible to define this as a locale variable inside of `deleteNonShuffleFiles` in order to reduce its scope to be as small as possible and make it clear that this is only used within `deleteNonShuffleFiles`?